python/tank/deploy/tank_commands/get_entity_commands.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from .action_base import Action
+from ...errors import TankError
+from ..util import execute_toolkit_command, SubprocessCalledProcessError
+
+import itertools
+import operator
+import os
+import sys
+
+
+class GetEntityCommandsAction(Action):
+    """
+    Gets the commands that can be launched on certain entities for another
+    pipeline configuration.
+
+    This is done by calling the tank command on the other pipeline
+    configuration and asking for its cached entity commands (or ask to update
+    its cache beforehand if needed).
+
+    It is used like this:
+    >>> import tank
+    # create our command object
+    >>> cmd = tank.get_command("get_entity_commands")
+    # get the commands for tasks, but could mix and match with any other types
+    >>> tasks = [("Task", 1234), ("Task", 1235)]
+    >>> commands_by_task = cmd.execute({"configuration_path": "/my/pc/path",
+    >>>                                 "entities": tasks})
+    # extract the commands of a specific task
+    >>> commands = commands_by_task[tasks[0]]
+    """
+
+    # Error codes that can be returned by the toolkit command that gets invoked
+    _ERROR_CODE_CACHE_OUT_OF_DATE = 1
```

Comment:
Minor: We tend to write this `(_ERROR_CODE_CACHE_OUT_OF_DATE, _ERROR_CODE_CACHE_NOT_FOUND) = range(2)`. I think that's a common recipe in python.